rslib/src/import_export/package/apkg/import.rs

```diff
@@ -393,21 +398,32 @@ impl<'a> Context<'a> {
 
     fn import_cards(&mut self) -> Result<()> {
         let existing = self.target_col.storage.all_cards_as_nid_and_ord()?;
+        let collection_delta = self.collection_delta()?;
+        let version = self.target_col.scheduler_info()?.version;
         for mut card in mem::take(&mut self.data.cards) {
-            if self.conflicting_notes.contains(&card.note_id) {
-                continue;
-            }
-            self.remap_note_id(&mut card);
-            if existing.contains(&(card.note_id, card.template_idx)) {
+            if !self.conflicting_notes.contains(&card.note_id) {
+                self.remap_note_id(&mut card);
+                if !existing.contains(&(card.note_id, card.template_idx)) {
+                    self.remap_deck_id(&mut card);
+                    card.shift_collection_relative_dates(collection_delta);
+                    card.maybe_remove_from_filtered_deck(version);
+                    self.add_card(&mut card)?;
+                }
                 // TODO: maybe update
-                continue;
             }
-            self.remap_deck_id(&mut card);
-            // TODO: adjust collection-relative due times
-            // TODO: remove cards from filtered decks
-            let old_id = mem::take(&mut card.id);
-            self.target_col.add_card(&mut card)?;
-            self.remapped_cards.insert(old_id, card.id);
+        }
+        Ok(())
+    }
+
+    /// The number of days the source collection is ahead of the target collection.
+    fn collection_delta(&mut self) -> Result<i32> {
+        Ok(self.data.days_elapsed as i32 - self.target_col.timing_today()?.days_elapsed as i32)
+    }
+
+    fn add_card(&mut self, card: &mut Card) -> Result<()> {
+        card.usn = self.usn;
+        if self.target_col.add_card_if_unique_undoable(card)? {
+            self.added_cards.insert(card.id);
```

Comment:
The original code also does 'insert or ignore', but if I've read it correctly, it is first giving the card a unique id, so theoretically the conflict case should never happen for cards, and the if_unique here makes things a bit more confusing.

Reply:
Yes, I've messed this up. So many objects, all with different equality checks and merge strategies. 🙂

Reply:
Yep, it's a mess - can you tell why this ended up being the last pylib area to get ported to Rust? :-)